keras_cv/layers/regularization/dropblock_2d.py

```diff
@@ -0,0 +1,191 @@
+
+        One must remember, that DropBlock operation is random, so bigger or smaller
+        patches can be dropped.
+    """
+
+    def __init__(
+        self,
+        keep_probability=0.9,
+        dropblock_size=7,
+        data_format=None,
+        seed=None,
+        name=None,
+    ):
+        super().__init__(seed=seed, name=name, force_generator=True)
+        if not dropblock_size > 0:
+            raise ValueError(
+                f"dropblock_size must be greater than 0. Received: {dropblock_size}"
+            )
+        if not 0.0 <= keep_probability <= 1.0:
+            raise ValueError(
+                f"keep_probability must be a number between 0 and 1. "
+                f"Received: {keep_probability}"
+            )
+
+        self._keep_probability = keep_probability
+        self._dropblock_size = dropblock_size
+        self._data_format = conv_utils.normalize_data_format(data_format)
+
+    def call(self, x, training=False):
+        if not training or self._keep_probability == 1.0:
+            return x
+
+        if self._data_format == "channels_last":
+            _, height, width, _ = x.get_shape().as_list()
+        else:
+            _, _, height, width = x.get_shape().as_list()
+
+        total_size = width * height
+        dropblock_size = min(self._dropblock_size, width, height)
+
+        # Seed_drop_rate is the gamma parameter of DropBlock.
+        seed_drop_rate = (
+            (1.0 - self._keep_probability)
+            * total_size
+            / dropblock_size**2
+            / ((width - self._dropblock_size + 1) * (height - self._dropblock_size + 1))
+        )
+
+        # Forces the block to be inside the feature map.
+        w_i, h_i = tf.meshgrid(tf.range(width), tf.range(height))
+        valid_block = tf.logical_and(
+            tf.logical_and(
+                w_i >= int(dropblock_size // 2), w_i < width - (dropblock_size - 1) // 2
+            ),
+            tf.logical_and(
+                h_i >= int(dropblock_size // 2), h_i < width - (dropblock_size - 1) // 2
+            ),
+        )
+
+        if self._data_format == "channels_last":
+            valid_block = tf.reshape(valid_block, [1, height, width, 1])
+        else:
+            valid_block = tf.reshape(valid_block, [1, 1, height, width])
+
+        randnoise = self._random_generator.random_uniform(tf.shape(x), dtype=tf.float32)
+        valid_block = tf.cast(valid_block, dtype=tf.float32)
+        seed_keep_rate = tf.cast(1 - seed_drop_rate, dtype=tf.float32)
+        block_pattern = (1 - valid_block + seed_keep_rate + randnoise) >= 1
+        block_pattern = tf.cast(block_pattern, dtype=tf.float32)
+
+        if self._data_format == "channels_last":
+            ksize = [1, self._dropblock_size, self._dropblock_size, 1]
+        else:
+            ksize = [1, 1, self._dropblock_size, self._dropblock_size]
+
+        block_pattern = -tf.nn.max_pool(
+            -block_pattern,
```

Comment:
Does this still work with negative numbers?  Maybe I am getting confused a bit here.

Add a comment explaining this line/what it does/how it works if you can.

Reply:
It seems that this double negative + max_pool is essentially min_pool - I will add a comment.

Reply:
Added comment explaining double negative.